src/Data/Array/ST.purs

```diff
@@ -146,12 +141,10 @@ thaw = copyImpl
 foreign import toAssocArray """
   function toAssocArray(arr) {
     return function(){
-      var as = [];
-      var i = -1;
-      var n = arr.length;
-      while(++i < n) {
+      var n = arr.length >>> 0;
+      var as = new Array(n);
+      for (var i = 0 >>> 0; i < n; i++)
```

Comment:
I'm just trying to hint the JIT that both values are unsigned:

    (-1) < ([].length)
    true
    (-1>>>0) < ([].length >>> 0)
    false

I guess this is completely useless since JITs are now quite smart.